ssl/ssl_lib.c

```diff
@@ -2372,7 +2372,11 @@ SSL_CTX *SSL_CTX_new(const SSL_METHOD *meth)
         goto err;
 
     ret->method = meth;
-    ret->min_proto_version = 0;
+    if (meth == TLS_method() || meth == TLS_server_method()
+        || meth == TLS_client_method())
+        ret->min_proto_version = TLS1_2_VERSION;
+    else
+        ret->min_proto_version = 0;
     ret->max_proto_version = 0;
```

Comment:
These really should not be function call tests - you should be looking at ret->method->version ... testing against TLS_ANY_VERSION etc ... this usage will drag in both client+server code (forced link reference) and it should be testing  the fields that are set rather than checking specific method pointers match.